scanner.py

```diff
@@ -2410,31 +2410,45 @@ def cmd_Z_OFFSET_APPLY_PROBE(self, gcmd):
         if offset == 0:
             self.gcode.respond_info("Nothing to do: Z Offset is 0")
             return
-
+            
         if not self.model:
-            raise self.gcode.error("You must calibrate your model first, "
-                                   "use SCANNER_CALIBRATE.")
-
-        # We use the model code to save the new offset, but we can't actually
-        # apply that offset yet because the gcode_offset is still in effect.
-        # If the user continues to do stuff after this, the newly set model
-        # offset would compound with the gcode offset. To ensure this doesn't
-        # happen, we revert to the old model offset afterwards.
-        # Really, the user should just be calling `SAVE_CONFIG` now.
+            raise self.gcode.error("You must calibrate your model first.")
+         
         if self.calibration_method == "touch":
-            self.scanner_touch_config['z_offset'] += offset
-            configfile = self.printer.lookup_object('configfile')
-            configfile.set("scanner", "scanner_touch_z_offset", "%.3f" % self.scanner_touch_config['z_offset'])
-            gcmd.respond_info(f"Touch offset has been updated by {offset:.3f} to {self.scanner_touch_config['z_offset']:.3f}.\n"
-                    "You must run the SAVE_CONFIG command now to update the\n"
-                    "printer config file and restart the printer.")
-
+            newoffset = self.scanner_touch_config['z_offset']
+            newoffset += offset
+            if newoffset < 0:
+                self.scanner_touch_config['z_offset'] = newoffset
+                configfile = self.printer.lookup_object('configfile')
+                configfile.set("scanner", "scanner_touch_z_offset", "%.3f" % 0)
+                gcmd.respond_info(f"Touch offset attempted to update to {newoffset:.3f}.\n"
+                    "However it cannot be less than 0. So its been set to 0.\n"
+                    "Please check your printers calibration and try again.")
+                return
+                                   
+        
+            else:
+                # We use the model code to save the new offset, but we can't actually
+                # apply that offset yet because the gcode_offset is still in effect.
+                # If the user continues to do stuff after this, the newly set model
+                # offset would compound with the gcode offset. To ensure this doesn't
+                # happen, we revert to the old model offset afterwards.
+                # Really, the user should just be calling `SAVE_CONFIG` now.
+                if self.calibration_method == "touch":
```

Comment:
We already check `self.calibration_method == "touch"` before this, on line 2417 🤔